### PR TITLE
Add Breadcrumb to Create Dataset Page

### LINKED ISSUE #480: Add Breadcrumb to Create Dataset Page
On the Create Dataset Page, we need to add the breadcrumb, similar to how it is displayed on the Create Collection Page.
For example: 
Screenshot 2024-09-06 at 12 46 13 PM

### PR DESCRIPTION
**Pull request by ChengShi-1**
## What this PR does / why we need it:
In Create Dataset page, we don't have the breadcrumb for the page which is supposed to have.

## Which issue(s) this PR closes:

- Closes #480 

## Special notes for your reviewer:
Changes:
1. Add breadcrumb to the page
2. Create a createDatasetSkeleton for loading state
3. Add containers to multiple 'Create Dataset' text existing for E2E testing(Collection.spec.ts)

## Suggestions on how to test this:
1. Log in to the SPA
2. Locate and click on the "Add Data" button in the navigation bar
3. From the dropdown or menu that appears, select "New Dataset"
5. Check the breadcrumb trail at the top of the page. It should display: "Root > Create Dataset"

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
Yes, breadcrumb added.
image

## Is there a release notes update needed for this change?:
No
## Additional documentation:
No

### CODE REVIEW

File: src/sections/create-dataset/CreateDataset.tsx [hunk 1]
```diff
@@ -13,6 +13,10 @@ import { CollectionRepository } from '../../collection/domain/repositories/Colle
 import { useLoading } from '../loading/LoadingContext'
 import { ROOT_COLLECTION_ALIAS } from '../../collection/domain/models/Collection'
 
+import { BreadcrumbsGenerator } from '../shared/hierarchy/BreadcrumbsGenerator'
+import { useCollection } from '../collection/useCollection'
+import { PageNotFound } from '../page-not-found/PageNotFound'
+import { CreateDatasetSkeleton } from './CreateDatasetSkeleton'
```
Comment (g-saracca) [inline src/sections/create-dataset/CreateDataset.tsx:19]: Could you add an empty line between this import and the actual component?

File: tests/e2e-integration/e2e/sections/collection/Collection.spec.ts [hunk 2]
```diff
@@ -42,7 +42,9 @@ describe('Collection Page', () => {
       cy.findByText('New Dataset').should('be.visible').click({ force: true })
     })
     cy.wait(1000)
-    cy.findByText(/Create Dataset/i).should('exist')
+    cy.get(`.breadcrumb`)
```
Comment (g-saracca) [inline tests/e2e-integration/e2e/sections/collection/Collection.spec.ts:45]: You can discard these assertions of the breadcrumb in this test as they are already being asserted in the CreateDataset spec

**Review by g-saracca — CHANGES_REQUESTED** [review]
Looks good! Congrats on your first PR @ChengShi-1 ! 🎉  
Just some minor changes and if you could please add the change I mentioned in the core-spa channel about the upload-artifact version so we can see that the component tests run correctly.

**Review by g-saracca — DISMISSED** [review]
Looks good! approving! 🚀

**Review by ekraffmiller — CHANGES_REQUESTED** [review]
Looks great, just requesting a clean up of older code in the e2e test

**Review by ekraffmiller — APPROVED** [review]
thanks for removing the wait() calls,  approved!

**Pull request merged**